lib/modules/manager/swift/extract.ts

```diff
@@ -117,20 +119,27 @@ function getMatch(str: string, state: string | null): MatchResult | null {
   return result;
 }
 
-function getDepName(url: string | null): string | null {
+function parseUrl(
+  url: string | null,
+): { depName: string; datasource: string } | null {
   // istanbul ignore if
   if (!url) {
     return null;
   }
   try {
     const { host, pathname } = new URL(url);
```

Comment:
use detectPlatform util function, so we also support ghes and self hosted gitlab.


if self hosted the return the base url as registry url